merkledag/merkledag.go

```diff
@@ -184,6 +184,9 @@ func (ds *dagService) GetNodes(ctx context.Context, keys []key.Key) []NodeGetter
 			select {
 			case blk, ok := <-blkchan:
 				if !ok {
+					for _, p := range promises {
+						p.Fail(ErrNotFound)
+					}
```

Comment:
This fails **all** `NodeGetter`s, right? Why can't we just let the failing ones fail and return the ones that do arrive?